check-services.php

```diff
@@ -61,10 +66,59 @@
     }
 }
 
-$sql = 'SELECT * FROM `devices` AS D, `services` AS S WHERE S.device_id = D.device_id ' . $where . ' ORDER by D.device_id DESC';
+$sql = 'SELECT * FROM `devices` AS D'
+       .' INNER JOIN `services` AS S ON S.device_id = D.device_id AND D.disabled = 0 '.$where
+       .' LEFT JOIN `devices_attribs` as A ON D.device_id = A.device_id AND A.attrib_type = "override_icmp_disable"'
+       .' ORDER by D.device_id DESC;';
 
 foreach (dbFetchRows($sql) as $service) {
-    // Run the polling function
-    poll_service($service);
-} //end foreach
+    // Run the polling function if the associated device is up, "Disable ICMP Test" option is not enabled,
+    // or service hostname/ip is different from associated device
+    if ($service['status'] === "1" || ($service['status'] === '0' && $service['status_reason'] === 'snmp') ||
+        $service['attrib_value'] === 'true' || ($service['service_ip'] !== $service['hostname'] &&
+        $service['service_ip'] !== inet6_ntop($service['ip']) )) {
+        // Mark service check as enabled if it was disabled previosly because device was down
```

Comment:
Just a spelling mistake on previosly.

Reply:
Agree about documentation. Will write something up during the weekend or on Monday, busy Friday.

I mark service as disabled in the `service_disabled` column in order to use it as a logical condition for writing to eventlog. It would write to eventlog only on first loop while the device is down.